docs/on_chain_views.md

```diff
@@ -0,0 +1,179 @@
+
+**Example of a contract script that declares multiple views:**
+
+```
+parameter nat ;
+storage nat ;
+code { CAR ; NIL operation ; PAIR } ;
+view "add" nat nat { UNPAIR ; ADD } ;
+view "id" nat (pair nat nat) { } ;
+view "test_failwith" nat (pair nat nat) { FAILWITH } ;
+view "step_constants" unit (pair (pair mutez mutez) (pair (pair address address) address ))
+     { DROP ;
+       SOURCE;
+       SENDER;
+       SELF_ADDRESS;
+       PAIR;
+       PAIR;
+       BALANCE;
+       AMOUNT;
+       PAIR;
+       PAIR;
+     } ;
+
+view "succ" (pair nat address) nat
+     { CAR;
+       UNPAIR;
+       PUSH nat 1; ADD;
+       PAIR;
+       DUP; CDR; SWAP;
+       VIEW "is_twenty" nat; ASSERT_SOME;
+     } ;
+view "is_twenty" (pair nat address) nat
+     {
+       CAR;
+       DUP;
+       CAR;
+       PUSH nat 20 ;
+       COMPARE;
+       EQ ;
+       IF { CAR; }
+          { DUP; CDR; SWAP; VIEW "succ" nat; ASSERT_SOME }
+     } ;
+view "fib" nat nat
+     {
+       CAR;
+       DUP;
+       PUSH nat 0 ;
+       COMPARE ;
+       EQ ;
+       IF { }
+          { DUP;
+            PUSH nat 1;
+            COMPARE;
+            EQ;
+            IF { }
+               { DUP;
+                 PUSH nat 1; SWAP; SUB; ABS;
+                 SELF_ADDRESS;
+                 SWAP;
+                 VIEW "fib" nat;
+                 IF_SOME { SWAP;
+                           PUSH nat 2; SWAP; SUB; ABS;
+                           SELF_ADDRESS;
+                           SWAP;
+                           VIEW "fib" nat;
+                           IF_SOME { ADD; } { FAIL }
+                         }
+                         { FAIL };
+               }
+          }
+     }
+```  
+  
+## Calling an on-chain view
+
+Views are meant to be called by a contract using the Michelson Instruction `View` followed by the view name and its result type.
+
+**Example of a contract script having a call to a view named `fib`:**
+
+```
+parameter (pair nat address) ;
+storage nat;
+code {
+       CAR;
+       UNPAIR;
+       VIEW "fib" nat;
+       IF_SOME {NIL operation ; PAIR;} { FAIL }
+     }
+```
+
+**Example of calling a contract entrypoint that makes a call to a view using Taquito:**
+
+The following live code example shows a contract (`contractCallFib`) calling the view `fib` of another contract (`contractTopLevelViews`). 
+
+The example first shows the initial storage of the contract `contractCallFib`. It calls the default entry point of `contractCallFib` with the value of its storage + 1 and the address of the contract `contractTopLevelViews`. A call is made the `fib` view of `contractTopLevelViews` with the `storage + 1` as argument. The view returns the value of the Fibonacci sequence at the position represented by `storage + 1`. The storage of `contractCallFib` is updated to the result of the view.
```

Comment:
"A call is made the "  ->  "A call is made to the .."

Reply:
Fixed, thanks!